streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableOuterJoin.java

```diff
@@ -126,16 +136,33 @@ public void init(final ProcessorContext context) {
         }
 
         @Override
-        public R get(final K key) {
+        public ValueAndTimestamp<R> get(final K key) {
             R newValue = null;
-            final V1 value1 = valueGetter1.get(key);
-            final V2 value2 = valueGetter2.get(key);
+            long resultTimestamp = -1;
```

Comment:
nit: I'd suggest use a constant instead of hard-coded `-1`: we can reuse RecordQueue.UNKNOWN e.g.

Reply:
It's a POC :)